File: 2022/Day_01/Python/solution.py
"""Solution to Advent of Code 2022 Day 1: Counting Calories."""


def process_elves(calorie_file: str) -> list:
    """Go through our input to create a list of calorie sums."""
    elf_sum = 0
    calorie_list = []
    with open(calorie_file) as input_file:
        elf_calories = input_file.readlines()
        for this_line in elf_calories:
            if this_line != "\n":
                elf_sum += int(this_line)
            else:
                calorie_list.append(elf_sum)
                elf_sum = 0
    # account for the final elf
    calorie_list.append(elf_sum)
    return calorie_list


def find_top_3_calorie_sum(elf_cals: list) -> int:
    """Take in a list of elf calories, sort, and sum the top 3."""
    elf_cals.sort(reverse=True)
    return elf_cals[0] + elf_cals[1] + elf_cals[2]


if __name__ == "__main__":
    elf_calories = process_elves("../input.txt")
    max_calorie_elf = max(elf_calories)
    print(f"The elf with the most calories is carrying {max_calorie_elf} calories.")
    top_three_cals = find_top_3_calorie_sum(elf_calories)
    print(f"The top 3 elves are carrying {top_three_cals} calories.")

File: 2022/Day_01/Ruby/solution.rb
# Solution to 2022 Day 01 - Calorie Counting

def input_per_line(file)
  File.readlines(file, chomp:true)
end

elf_calorie_list = input_per_line("../input.txt")
calorie_per_elf = elf_calorie_list.map(&:to_i).slice_when {|i, j| j==0}.to_a
elf_sum = []
calorie_per_elf.each do |item|
  elf_sum.append(item.sum)
end
elf_sum.sort!.reverse!

puts "The elf with the highest amount of calories is carrying #{elf_sum[0]} calories. "

puts "The top 3 elves with the highest amount of calories are carrying #{elf_sum[0]+elf_sum[1]+elf_sum[2]} calories. "